pkg/v1/tkg/managementcomponents/management_component_install.go

```
@@ -49,18 +60,183 @@ type ManagementComponentsInstallOptions struct {
+	}
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	return nil
+}
+
+func DeleteAddonSecret(clusterName, addonName, namespace string, clusterClient clusterclient.Client) error {
+	addonSecret := &corev1.Secret{}
+	addonSecret.Name = generateAddonSecretName(clusterName, addonName)
+	addonSecret.Namespace = constants.TkgNamespace
+	err := clusterClient.GetResource(addonSecret, addonSecret.Name, addonSecret.Namespace, nil, nil)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if controllerutil.ContainsFinalizer(addonSecret, addonFinalizer) {
+		controllerutil.RemoveFinalizer(addonSecret, addonFinalizer)
+	}
+	err = clusterClient.UpdateResource(addonSecret, addonSecret.Name, addonSecret.Namespace)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	err = clusterClient.DeleteResource(addonSecret)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete addon addonSecret %s", addonSecret)
+	}
+	return nil
+}
+
 // InstallManagementComponents installs the management component to cluster
 func InstallManagementComponents(mcip *ManagementComponentsInstallOptions) error {
 	clusterClient, err := clusterclient.NewClient(mcip.ClusterOptions.Kubeconfig, mcip.ClusterOptions.Kubecontext, clusterclient.Options{})
 	if err != nil {
 		return errors.Wrap(err, "unable to get cluster client")
 	}
+	clusterName, err := clusterClient.GetCurrentClusterName(mcip.ClusterOptions.Kubecontext)
+	if err != nil {
+		return errors.Wrap(err, "unable to get cluster name")
+	}
+	if err != nil {
+		return err
+	}
 
-	// create package client
-	pkgClient, err := tkgpackageclient.NewTKGPackageClientForContext(mcip.ClusterOptions.Kubeconfig, mcip.ClusterOptions.Kubecontext)
+	err = PauseAddonLifecycleManagement(clusterName, addonsManagerName, constants.TkgNamespace, clusterClient)
+	if err != nil {
+		panic(err)
```

Why should this one panic instead of returning an error?

---

fixed